builtin/providers/aws/data_source_aws_route_table.go

```diff
@@ -0,0 +1,79 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsRouteTable() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsRouteTableRead,
+
+		Schema: map[string]*schema.Schema{
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"filter": ec2CustomFiltersSchema(),
+
+			"id": {
```

Comment:
This is not needed (as per the earlier comment, you do not to set `id` explicitly - this is done via the `d.SetId()`.